libyul/optimiser/ExpressionSimplifier.h

```diff
@@ -33,22 +33,20 @@ struct Dialect;
  * The component will work best if the code is in SSA form, but
  * this is not required for correctness.
  *
+ * It tracks the current values of variables using the DataFlowAnalyzer
+ * and takes them into account for replacements.
```

Comment:
Does that mean it always takes the latest reference in scope into account?

Reply:
The DataFlowAnalyzer tracks the latest assigned value (but only if it is a movable expression) including all variables referenced in that expression. If one of them is reassigned, all knowledge about variables whose current value reference that variable is cleared. The same thing is done before e.g. a for loop is entered: If `x` is assigned inside the for loop, all variables whose current value references `x` are cleared.

Reply:
Ok!